src/engines/nwn2/roster.cpp

```diff
@@ -34,29 +36,44 @@ namespace Engines {
 
 namespace NWN2 {
 
-Roster::Roster() : _lastRetrieved(UINT32_MAX) {
+Roster::Roster() : _partyLimit(3), _lastRetrieved(UINT32_MAX) {
 	load();
 }
 
 Roster::~Roster() {
 }
 
-/** Add a creature by template name to the roster list. */
-bool Roster::addRosterMemberByTemplate(Common::UString name, Common::UString cTemplate) {
+uint32 Roster::getRosterNPCPartyLimit() {
+	return _partyLimit;
+}
+
+void Roster::setRosterNPCPartyLimit(uint32 limit) {
+	_partyLimit = limit;
+}
+
+/**
+ * Add a creature by template name to the roster list.
+ * The 'unitTest' flag is required for unit testing
+ * because the UTC files will not be available.
+ */
+bool Roster::addRosterMemberByTemplate(Common::UString name, Common::UString cTemplate, bool unitTest) {
```

Comment:
I'm not really all that fond of that unitTest param :/

Reply:
Is there an alternative other than to not run the test? Like a unit test state function I can call?

Reply:
Not really, no.

I can't think of a better solution. We can't mock the ResourceManager either.
Maybe some way to tell the ResourceManager to lie? An anti-blacklist of sorts?

Probably a discussion for the future. Let's keep that param for now, I guess. Please mark it with a "HACK" comment.

Reply:
Okay, I just had a thought to just use a special token in the name field, such as "[GTEST]". If the ResMan lookup fails, then it can check for the token. Thanks.